app/main.py

```diff
@@ -1,2 +1,56 @@
+from typing import Any
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.size = 0
+        self.hash_table = self.create_hash_table()
+        self.load_factor = self.calculate_load_factor()
+
+    def create_hash_table(self) -> list:
+        return [None for _ in range(self.capacity)]
+
+    def calculate_load_factor(self) -> int:
+        return int(self.capacity * (2 / 3))
+
+    def __len__(self) -> int:
+        return self.size
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        hash_key = hash(key)
+        position = hash_key % self.capacity
+        while True:
+            if not self.hash_table[position]:
+                self.hash_table[position] = (key, value, hash_key)
+                self.size += 1
+                if self.size >= self.load_factor:
+                    self.resize()
+                break
+            elif self.hash_table[position][0] == key:
+                self.hash_table[position] = (key, value, hash_key)
+                break
```

Comment:
You can write you while loop like this :
`while not self.hash_table[position]:`
That way your code will bw less nested imho. :)